File: app/src/main/java/com/lz233/onetext/BaseActivity.java
package com.lz233.onetext;

import android.app.Activity;
import android.content.Context;
import android.content.SharedPreferences;
import android.content.res.Configuration;
import android.os.Bundle;
import android.os.Environment;
import android.view.View;
import android.view.WindowManager;

import androidx.annotation.Nullable;
import androidx.appcompat.app.AppCompatActivity;
import androidx.appcompat.app.AppCompatDelegate;

import io.github.inflationx.calligraphy3.CalligraphyConfig;
import io.github.inflationx.calligraphy3.CalligraphyInterceptor;
import io.github.inflationx.viewpump.ViewPump;
import io.github.inflationx.viewpump.ViewPumpContextWrapper;
public abstract class BaseActivity extends AppCompatActivity {
    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        SharedPreferences sharedPreferences = getSharedPreferences("setting",MODE_PRIVATE);
        super.onCreate(savedInstanceState);
        //状态栏icon黑色
        int mode = getResources().getConfiguration().uiMode & Configuration.UI_MODE_NIGHT_MASK;
        if(mode == Configuration.UI_MODE_NIGHT_NO) {
            this.getWindow().getDecorView().setSystemUiVisibility(View.SYSTEM_UI_FLAG_LIGHT_STATUS_BAR);
        }
        //设置为miui主题
        //setMiuiTheme(BaseActivity.this,0,mode);
        getWindow().setFlags(
                WindowManager.LayoutParams.FLAG_TRANSLUCENT_NAVIGATION,
                WindowManager.LayoutParams.FLAG_TRANSLUCENT_NAVIGATION);
        //全局自定义字体
        ViewPump.init(ViewPump.builder()
                .addInterceptor(new CalligraphyInterceptor(
                        new CalligraphyConfig.Builder()
                                .setDefaultFontPath(sharedPreferences.getString("font_path",null))
                                .setFontAttrId(R.attr.fontPath)
                                .build()))
                .build());
    }
    @Override
    protected void attachBaseContext(Context newBase) {
        super.attachBaseContext(ViewPumpContextWrapper.wrap(newBase));
    }
    public static void setMiuiTheme(Activity act, int overrideTheme,int isnightmode) {
        int themeResId = 0;
        try {
            themeResId = act.getResources().getIdentifier("Theme.DayNight", "style", "miui");
        } catch (Throwable t) {}
        if (themeResId == 0) themeResId = act.getResources().getIdentifier((isnightmode == Configuration.UI_MODE_NIGHT_YES) ? "Theme.Dark" : "Theme.Light", "style", "miui");
        act.setTheme(themeResId);
        act.getTheme().applyStyle(overrideTheme, true);
    }
}

File: instant/src/main/java/com/lz233/onetext/activity/BaseActivity.kt
package com.lz233.onetext.activity

import android.content.Context
import android.content.SharedPreferences
import android.content.res.Configuration
import android.os.Build
import android.os.Bundle
import android.view.View
import android.view.ViewGroup
import android.view.WindowManager
import android.widget.LinearLayout
import android.widget.RelativeLayout
import androidx.appcompat.app.AppCompatActivity
import androidx.core.view.ViewCompat
import androidx.core.view.WindowInsetsCompat
import com.lz233.onetext.R

abstract class BaseActivity : AppCompatActivity() {
    protected lateinit var rootView: ViewGroup
    protected lateinit var sharedPreferences: SharedPreferences
    protected lateinit var editor: SharedPreferences.Editor
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        sharedPreferences = getSharedPreferences("setting", Context.MODE_PRIVATE)
        editor = sharedPreferences.edit()
        rootView = findViewById(android.R.id.content)
        //状态栏icon黑色
        val mode = resources.configuration.uiMode and Configuration.UI_MODE_NIGHT_MASK
        if (mode == Configuration.UI_MODE_NIGHT_NO) {
            window.decorView.systemUiVisibility = View.SYSTEM_UI_FLAG_LIGHT_STATUS_BAR or View.SYSTEM_UI_FLAG_LIGHT_NAVIGATION_BAR
        }
        rootView.setSystemUiVisibility(View.SYSTEM_UI_FLAG_LAYOUT_STABLE or View.SYSTEM_UI_FLAG_LAYOUT_HIDE_NAVIGATION)
        ViewCompat.setOnApplyWindowInsetsListener(rootView) { v: View?, insets: WindowInsetsCompat ->
            rootView.setPadding(insets.systemWindowInsetLeft, 0, insets.systemWindowInsetRight, 0)
            insets
        }
        if (Build.VERSION.SDK_INT < Build.VERSION_CODES.O) {
            window.setFlags(WindowManager.LayoutParams.FLAG_TRANSLUCENT_NAVIGATION, WindowManager.LayoutParams.FLAG_TRANSLUCENT_NAVIGATION)
        }
    }

    fun getNavigationBarHeight(): Int {
        val resources = this.resources
        val resourceId = resources.getIdentifier("navigation_bar_height", "dimen", "android")
        //Log.v("dbw", "Navi height:" + height);
        return resources.getDimensionPixelSize(resourceId)
    }

    fun fuckNav(last_layout: View) {
        val lp = last_layout.layoutParams as LinearLayout.LayoutParams
        lp.setMargins(resources.getDimensionPixelSize(R.dimen.layout_margin), 0, resources.getDimensionPixelSize(R.dimen.layout_margin), getNavigationBarHeight() + resources.getDimensionPixelSize(R.dimen.layout_margin))
        last_layout.layoutParams = lp
    }

    fun fuckNav(last_layout: View, left: Int, top: Int, right: Int, bottom: Int) {
        val lp = last_layout.layoutParams as RelativeLayout.LayoutParams
        lp.setMargins(left, top, right, bottom)
        last_layout.layoutParams = lp
    }
}